src/ai/backend/web/proxy.py

```diff
@@ -131,31 +131,31 @@ async def close_upstream(self) -> None:
 async def decrypt_payload(request: web.Request, handler) -> web.StreamResponse:
     request_headers = extra_config_headers.check(request.headers)
     secure_context = request_headers.get("X-BackendAI-Encoded", None)
-    if secure_context:
-        if not request.content:
-            request["payload"] = ""
-            return await handler(request)
-        config = request.app["config"]
-        scheme = config["service"]["force_endpoint_protocol"]
-        if scheme is None:
-            scheme = request.scheme
-        api_endpoint = f"{scheme}://{request.host}"
-        payload = await request.text()
-        initial_vector, real_payload = payload.split(":")
-        key = (base64.b64encode(api_endpoint.encode("ascii")).decode() + initial_vector * 2)[0:32]
-        crypt = AES.new(
-            bytes(key, encoding="utf8"), AES.MODE_CBC, bytes(initial_vector, encoding="utf8")
-        )
-        b64p = base64.b64decode(real_payload)
-        dec = unpad(crypt.decrypt(bytes(b64p)), 16)
-        result = dec.decode("UTF-8")
-        if not result:
-            request["payload"] = ""
+    try:
+        if secure_context:
+            if not request.content:
+                request["payload"] = ""
+                return await handler(request)
+            config = request.app["config"]
+            scheme = config["service"]["force_endpoint_protocol"]
+            if scheme is None:
+                scheme = request.scheme
+            api_endpoint = f"{scheme}://{request.host}"
+            payload = await request.text()
+            initial_vector, real_payload = payload.split(":")
```

Comment:
I'd like to move the beginning of `try` block to here, as the core parsing logic starts here.